irc/handlers.go

```diff
@@ -1792,6 +1784,9 @@ func operHandler(server *Server, client *Client, msg ircmsg.IrcMessage, rb *Resp
 
 	server.snomasks.Send(sno.LocalOpers, fmt.Sprintf(ircfmt.Unescape("Client opered up $c[grey][$r%s$c[grey], $r%s$c[grey]]"), client.nickMaskString, client.operName))
 
+	// increase oper count
+	server.stats.ChangeOperators(1)
```

Comment:
Sorry, one more thing: is there a reason this can't be done inside `applyUserModeChanges`, like the corresponding decrement operation?

Reply:
When the oper modes are added; they are "forced" (assuming so people can't add +oO to themselves) and I had some troubles making it work inside the mode handler - it seemed it never got triggered for me inside of it. Suggestions for location is appreciated. :smile:

Reply:
Yeah here's the best place for this change

Reply:
Yeah, I agree. The odd thing is if I put the logic inside the modes.Add case, it does not seem like it gets evaluated; 

Snippet:
```go
case modes.Bot, modes.Invisible, modes.WallOps, modes.UserRoleplaying, modes.Operator, modes.LocalOperator, modes.RegisteredOnly:
	switch change.Op {
		case modes.Add:
			if !force && (change.Mode == modes.Operator || change.Mode == modes.LocalOperator) {
				continue
			}
			if change.Mode == modes.Operator || change.Mode == modes.LocalOperator {
				client.server.stats.ChangeOperators(1)
			}
```

Same issue arises no matter where I stick the code inside modes.Add - be that above the first check for whether it was forced, or not.

Reply:
This seems to be due to the following snippet in operHandler

```go

	applied = append(applied, modes.ModeChange{
		Mode: modes.Operator,
		Op:   modes.Add,
	})
```

which bypasses applyUserModeChanges altogether.

Reply:
Argh, modes is a mess --- we'll leave this as-is and then clean up the whole system in #255. Thanks for the excellent PR!